PR: https://github.com/odoo/odoo/pull/236895

From: 3852885f37aa691149f0be7d05d0ef9add91c163
From: Stéphane Diez
Date: 2026-01-23 13:16:51

Breaking data model changes scores: del:6 + add:0, change matches:
-    product_uom_id = fields.Many2one(
-    product_uom = fields.Many2one(
-    product_uom_id = fields.Many2one(
-    product_uom = fields.Many2one(
-    product_uom_id = fields.Many2one(
-    product_uom_id = fields.Many2one(related='product_id.uom_id')

Total Changes: 402

[IMP] uom: rename logistic fields

Currently, the uom field can have 3 different names in logistic modules:
* uom_id
* product_uom_id
* product_uom

This is annoying when we need to use a uom field and we have to
check what is its name. So we decided to only keep uom_id as the
only name for basic uom fields.

task 5364113

Part-of: odoo/odoo#236895
Related: odoo/enterprise#100172
Related: odoo/upgrade#8959
Signed-off-by: Arnold Moyaux (arm) <arm@odoo.com>

================================= pseudo patch: =================================

--- a/addons/stock/models/product.py
+++ b/addons/stock/models/product.py
@@ -209,7 +209,7 @@ class ProductProduct(models.Model):
             domain_move_in_done = [('state', '=', 'done'), ('date', '>', to_date)] + domain_move_in_done
             domain_move_out_done = [('state', '=', 'done'), ('date', '>', to_date)] + domain_move_out_done
 
-            groupby = ['product_id', 'product_uom']
+            groupby = ['product_id', 'uom_id']
             for product, uom, quantity in Move._read_group(domain_move_in_done, groupby, ['quantity:sum']):
                 moves_in_res_past[product.id] += uom._compute_quantity(quantity, product.uom_id)
 
@@ -707,7 +707,7 @@ class ProductProduct(models.Model):
     def _update_uom(self, to_uom_id):
         for uom, product, moves in self.env['stock.move']._read_group(
             [('product_id', 'in', self.ids)],
-            ['product_uom', 'product_id'],
+            ['uom_id', 'product_id'],
             ['id:recordset'],
         ):
             if uom != product.product_tmpl_id.uom_id:
@@ -715,11 +715,11 @@ class ProductProduct(models.Model):
                 'than %(uom)s have already been used for this product, the change of unit of measure can not be done.'
                 'If you want to change it, please archive the product and create a new one.',
                 problem_uom=uom.name, uom=product.product_tmpl_id.uom_id.name))
-            moves.product_uom = to_uom_id
+            moves.uom_id = to_uom_id
 
         for uom, product, move_lines in self.env['stock.move.line']._read_group(
             [('product_id', 'in', self.ids)],
-            ['product_uom_id', 'product_id'],
+            ['uom_id', 'product_id'],
             ['id:recordset'],
         ):
             if uom != product.product_tmpl_id.uom_id:
@@ -727,7 +727,7 @@ class ProductProduct(models.Model):
                 'than %(uom)s have already been used for this product, the change of unit of measure can not be done.'
                 'If you want to change it, please archive the product and create a new one.',
                 problem_uom=uom.name, uom=product.product_tmpl_id.uom_id.name))
-            move_lines.product_uom_id = to_uom_id
+            move_lines.uom_id = to_uom_id
         return super()._update_uom(to_uom_id)
 
     def filter_has_routes(self):
@@ -1219,12 +1219,12 @@ class UomUom(models.Model):
                     " or are currently reserved."
                 )
                 if self.env['stock.move'].sudo().search_count([
-                    ('product_uom', 'in', changed.ids),
+                    ('uom_id', 'in', changed.ids),
                     ('state', 'not in', ('cancel', 'done'))
                 ]):
                     raise UserError(error_msg)
                 if self.env['stock.move.line'].sudo().search_count([
-                    ('product_uom_id', 'in', changed.ids),
+                    ('uom_id', 'in', changed.ids),
                     ('state', 'not in', ('cancel', 'done')),
                 ]):
                     raise UserError(error_msg)

--- a/addons/stock/models/stock_location.py
+++ b/addons/stock/models/stock_location.py
@@ -353,7 +353,7 @@ class StockLocation(models.Model):
                         ('product_id', '=', product.id),
                         ('location_dest_id', 'in', locations.ids),
                         ('state', 'not in', ['draft', 'done', 'cancel'])
-                    ], ['location_dest_id'], ['quantity:array_agg', 'product_uom_id:recordset'])
+                    ], ['location_dest_id'], ['quantity:array_agg', 'uom_id:recordset'])
                     quant_data = self.env['stock.quant']._read_group([
                         ('product_id', '=', product.id),
                         ('location_id', 'in', locations.ids),

--- a/addons/stock/models/stock_lot.py
+++ b/addons/stock/models/stock_lot.py
@@ -46,7 +46,7 @@ class StockLot(models.Model):
         domain=("[('tracking', '!=', 'none'), ('is_storable', '=', True)] +"
             " ([('product_tmpl_id', '=', context['default_product_tmpl_id'])] if context.get('default_product_tmpl_id') else [])"),
         required=True, check_company=True, tracking=True)
-    product_uom_id = fields.Many2one(
+    uom_id = fields.Many2one(
         'uom.uom', 'Unit',
         related='product_id.uom_id')
     quant_ids = fields.One2many('stock.quant', 'lot_id', 'Quants', readonly=True)

--- a/addons/stock/models/stock_move.py
+++ b/addons/stock/models/stock_move.py
@@ -64,9 +64,9 @@ class StockMove(models.Model):
              "Changing this quantity on assigned moves affects "
              "the product reservation, and should be done with care.")
     allowed_uom_ids = fields.Many2many('uom.uom', compute='_compute_allowed_uom_ids')
-    product_uom = fields.Many2one(
+    uom_id = fields.Many2one(
         'uom.uom', "Unit", required=True, domain="[('id', 'in', allowed_uom_ids)]",
-        compute="_compute_product_uom", store=True, readonly=False, precompute=True,
+        compute="_compute_uom_id", store=True, readonly=False, precompute=True,
     )
     # TDE FIXME: make it stored, otherwise group will not work
     product_tmpl_id = fields.Many2one(
@@ -201,15 +201,15 @@ class StockMove(models.Model):
 
     _product_location_index = models.Index("(product_id, location_id, location_dest_id, company_id, state)")
 
-    @api.depends('product_id', 'product_id.uom_id', 'product_id.uom_ids', 'product_id.seller_ids', 'product_id.seller_ids.product_uom_id')
+    @api.depends('product_id', 'product_id.uom_id', 'product_id.uom_ids', 'product_id.seller_ids', 'product_id.seller_ids.uom_id')
     def _compute_allowed_uom_ids(self):
         for move in self:
-            move.allowed_uom_ids = move.product_id.uom_id | move.product_id.uom_ids | move.sudo().product_id.seller_ids.product_uom_id
+            move.allowed_uom_ids = move.product_id.uom_id | move.product_id.uom_ids | move.sudo().product_id.seller_ids.uom_id
 
     @api.depends('product_id')
-    def _compute_product_uom(self):
+    def _compute_uom_id(self):
         for move in self:
-            move.product_uom = move.product_id.uom_id.id
+            move.uom_id = move.product_id.uom_id.id
 
     @api.depends('picking_id.location_id', 'is_scrap')
     def _compute_location_id(self):
@@ -366,7 +366,7 @@ class StockMove(models.Model):
                 if move.inventory_name:
                     move.reference = move.inventory_name
                 else:
-                    move.reference = _('Product Quantity Confirmed') if float_is_zero(move.quantity, precision_rounding=move.product_uom.rounding) else _('Product Quantity Updated')
+                    move.reference = _('Product Quantity Confirmed') if float_is_zero(move.quantity, precision_rounding=move.uom_id.rounding) else _('Product Quantity Updated')
                     if move.create_uid and move.create_uid.id != SUPERUSER_ID:
                         move.reference += f' ({move.create_uid.display_name})'
             else:
@@ -377,10 +377,10 @@ class StockMove(models.Model):
         for move in self:
             move.move_lines_count = len(move.move_line_ids)
 
-    @api.depends('product_id', 'product_uom', 'product_uom_qty', 'state')
+    @api.depends('product_id', 'uom_id', 'product_uom_qty', 'state')
     def _compute_product_qty(self):
         for move in self:
-            move.product_qty = move.product_uom._compute_quantity(
+            move.product_qty = move.uom_id._compute_quantity(
                 move.product_uom_qty, move.product_id.uom_id, rounding_method='HALF-UP')
 
     @api.depends('picking_id.partner_id')
@@ -405,10 +405,10 @@ class StockMove(models.Model):
         self.ensure_one()
         quantity = 0
         for move_line in self.move_line_ids:
-            quantity += move_line.product_uom_id._compute_quantity(move_line.quantity, self.product_uom, round=False)
+            quantity += move_line.uom_id._compute_quantity(move_line.quantity, self.uom_id, round=False)
         return quantity
 
-    @api.depends('move_line_ids.quantity', 'move_line_ids.product_uom_id')
+    @api.depends('move_line_ids.quantity', 'move_line_ids.uom_id')
     def _compute_quantity(self):
         """ This field represents the sum of the move lines `quantity`. It allows the user to know
         if there is still work to do.
@@ -430,12 +430,12 @@ class StockMove(models.Model):
 
             data = self.env['stock.move.line']._read_group(
                 [('id', 'in', list(move_lines_ids))],
-                ['move_id', 'product_uom_id'], ['quantity:sum']
+                ['move_id', 'uom_id'], ['quantity:sum']
             )
             sum_qty = defaultdict(float)
-            for move, product_uom, qty_sum in data:
-                uom = move.product_uom
-                sum_qty[move.id] += product_uom._compute_quantity(qty_sum, uom, round=False)
+            for move, uom_id, qty_sum in data:
+                uom = move.uom_id
+                sum_qty[move.id] += uom_id._compute_quantity(qty_sum, uom, round=False)
 
             for move in self:
                 move.quantity = sum_qty[move.id]
@@ -448,16 +448,16 @@ class StockMove(models.Model):
             for ml in reversed(move.move_line_ids.sorted('id')):
                 if self.env.context.get('unreserve_unpicked_only') and ml.picked:
                     continue
-                if move.product_uom.is_zero(quantity):
+                if move.uom_id.is_zero(quantity):
                     break
-                qty_ml_dec = min(ml.quantity, ml.product_uom_id._compute_quantity(quantity, ml.product_uom_id, round=False))
-                if ml.product_uom_id.is_zero(qty_ml_dec):
+                qty_ml_dec = min(ml.quantity, ml.uom_id._compute_quantity(quantity, ml.uom_id, round=False))
+                if ml.uom_id.is_zero(qty_ml_dec):
                     continue
-                if ml.product_uom_id.compare(ml.quantity, qty_ml_dec) == 0 and ml.state not in ['done', 'cancel']:
+                if ml.uom_id.compare(ml.quantity, qty_ml_dec) == 0 and ml.state not in ['done', 'cancel']:
                     mls_to_unlink.add(ml.id)
                 else:
                     ml.quantity -= qty_ml_dec
-                quantity -= move.product_uom._compute_quantity(qty_ml_dec, move.product_uom, round=False)
+                quantity -= move.uom_id._compute_quantity(qty_ml_dec, move.uom_id, round=False)
             self.env['stock.move.line'].browse(mls_to_unlink).unlink()
 
         def _process_increase(move, quantity):
@@ -475,9 +475,9 @@ Please change the quantity done or the rounding precision in your settings.""",
                              product=move.product_id.display_name))
                 continue
             delta_qty = move.quantity - move._quantity_sml()
-            if move.product_uom.compare(delta_qty, 0) > 0:
+            if move.uom_id.compare(delta_qty, 0) > 0:
                 _process_increase(move, delta_qty)
-            elif move.product_uom.compare(delta_qty, 0) < 0:
+            elif move.uom_id.compare(delta_qty, 0) < 0:
                 _process_decrease(move, abs(delta_qty))
         if err:
             raise UserError('\n'.join(err))
@@ -545,7 +545,7 @@ Please change the quantity done or the rounding precision in your settings.""",
             else:
                 free_qty = 0.0
             if move.state == 'assigned':
-                move.forecast_availability = move.product_uom._compute_quantity(
+                move.forecast_availability = move.uom_id._compute_quantity(
                     move.quantity, move.product_id.uom_id, rounding_method='HALF-UP')
                 continue
             elif move.state == 'draft' and float_compare(free_qty, move.product_qty, precision_rounding=move.product_id.uom_id.rounding) >= 0:
@@ -626,7 +626,7 @@ Please change the quantity done or the rounding precision in your settings.""",
                         move_line = mls_without_lots[:1]
                         move_lines_commands.append(Command.update(move_line.id, {
                             'lot_id': lot.id,
-                            'product_uom_id': move.product_id.uom_id.id if move.product_id.tracking == 'serial' else move.product_uom.id,
+                            'uom_id': move.product_id.uom_id.id if move.product_id.tracking == 'serial' else move.uom_id.id,
                             'quantity': 1 if move.product_id.tracking == 'serial' else move.quantity,
                         }))
                         mls_without_lots -= move_line
@@ -637,7 +637,7 @@ Please change the quantity done or the rounding precision in your settings.""",
                         else:
                             move_line_vals = self._prepare_move_line_vals(quantity=0)
                             move_line_vals['lot_id'] = lot.id
-                        move_line_vals['product_uom_id'] = move.product_id.uom_id.id
+                        move_line_vals['uom_id'] = move.product_id.uom_id.id
                         move_line_vals['quantity'] = 1
                         move_lines_commands.append((0, 0, move_line_vals))
                 else:
@@ -656,7 +656,7 @@ Please change the quantity done or the rounding precision in your settings.""",
             elif move.picking_type_id.reservation_method == 'manual':
                 move.reservation_date = False
 
-    @api.depends('product_uom', 'move_orig_ids', 'move_dest_ids', 'move_orig_ids.packaging_uom_id', 'move_dest_ids.packaging_uom_id')
+    @api.depends('uom_id', 'move_orig_ids', 'move_dest_ids', 'move_orig_ids.packaging_uom_id', 'move_dest_ids.packaging_uom_id')
     def _compute_packaging_uom_id(self):
         for move in self:
             if move.move_orig_ids.packaging_uom_id:
@@ -664,13 +664,13 @@ Please change the quantity done or the rounding precision in your settings.""",
             elif move.move_dest_ids.packaging_uom_id:
                 move.packaging_uom_id = move.move_dest_ids[0].packaging_uom_id
             else:
-                move.packaging_uom_id = move.product_uom
+                move.packaging_uom_id = move.uom_id
 
     @api.depends('product_uom_qty', 'packaging_uom_id')
     def _compute_packaging_uom_qty(self):
         for move in self:
             if move.packaging_uom_id:
-                move.packaging_uom_qty = move.product_uom._compute_quantity(move.product_uom_qty, move.packaging_uom_id)
+                move.packaging_uom_qty = move.uom_id._compute_quantity(move.product_uom_qty, move.packaging_uom_id)
 
     @api.depends(
         'has_tracking',
@@ -768,15 +768,15 @@ Please change the quantity done or the rounding precision in your settings.""",
         if 'quantity' in vals:
             if any(move.state == 'cancel' for move in self):
                 raise UserError(_('You cannot change a cancelled stock move, create a new line instead.'))
-        if 'product_uom' in vals and any(move.state == 'done' for move in self) and not self.env.context.get('skip_uom_conversion'):
+        if 'uom_id' in vals and any(move.state == 'done' for move in self) and not self.env.context.get('skip_uom_conversion'):
             raise UserError(_('You cannot change the UoM for a stock move that has been set to \'Done\'.'))
         if 'product_uom_qty' in vals:
             for move in self.filtered(lambda m: m.state not in ('done', 'draft') and m.picking_id):
-                if move.product_uom.compare(vals['product_uom_qty'], move.product_uom_qty):
+                if move.uom_id.compare(vals['product_uom_qty'], move.product_uom_qty):
                     self.env['stock.move.line']._log_message(move.picking_id, move, 'stock.track_move_template', vals)
             if self.env.context.get('do_not_unreserve') is None:
                 move_to_unreserve = self.filtered(
-                    lambda m: m.state not in ['draft', 'done', 'cancel'] and m.product_uom.compare(m.quantity, vals.get('product_uom_qty')) == 1
+                    lambda m: m.state not in ['draft', 'done', 'cancel'] and m.uom_id.compare(m.quantity, vals.get('product_uom_qty')) == 1
                 )
                 move_to_unreserve._do_unreserve()
                 (self - move_to_unreserve).filtered(lambda m: m.state == 'assigned').write({'state': 'partially_available'})
@@ -1094,7 +1094,7 @@ Please change the quantity done or the rounding precision in your settings.""",
             vals_list.append({**default_vals,
                              **lot,
                              'location_dest_id': loc_dest.id,
-                             'product_uom_id': product.uom_id.id,
+                             'uom_id': product.uom_id.id,
                             })
         if default_vals.get('picking_type_id'):
             picking_type = self.env['stock.picking.type'].browse(default_vals['picking_type_id'])
@@ -1182,7 +1182,7 @@ Please change the quantity done or the rounding precision in your settings.""",
     def _prepare_merge_moves_distinct_fields(self):
         fields = [
             'product_id', 'price_unit', 'procure_method', 'location_id', 'location_dest_id', 'location_final_id',
-            'product_uom', 'restrict_partner_id', 'origin_returned_move_id',
+            'uom_id', 'restrict_partner_id', 'origin_returned_move_id',
             'propagate_cancel', 'description_picking', 'never_product_template_attribute_value_ids',
         ]
         if self.env['ir.config_parameter'].sudo().get_bool('stock.merge_only_same_date'):
@@ -1250,7 +1250,7 @@ Please change the quantity done or the rounding precision in your settings.""",
 
         moves_by_neg_key = defaultdict(lambda: self.env['stock.move'])
         # Need to check less fields for negative moves as some might not be set.
-        neg_qty_moves = self.filtered(lambda m: m.product_uom.compare(m.product_qty, 0.0) < 0)
+        neg_qty_moves = self.filtered(lambda m: m.uom_id.compare(m.product_qty, 0.0) < 0)
         # Detach their picking as they will either get absorbed or create a backorder, so no extra logs will be put in the chatter
         neg_qty_moves.picking_id = False
         excluded_fields = self._prepare_merge_negative_moves_excluded_distinct_fields()
@@ -1280,7 +1280,7 @@ Please change the quantity done or the rounding precision in your settings.""",
             for pos_move in moves_by_neg_key.get(neg_key(neg_move), []):
                 new_total_value = pos_move.product_qty * pos_move.price_unit + neg_move.product_qty * neg_move.price_unit
                 # If quantity can be fully absorbed by a single move, update its quantity and remove the negative move
-                if pos_move.product_uom.compare(pos_move.product_uom_qty, abs(neg_move.product_uom_qty)) >= 0:
+                if pos_move.uom_id.compare(pos_move.product_uom_qty, abs(neg_move.product_uom_qty)) >= 0:
                     pos_move.product_uom_qty += neg_move.product_uom_qty
                     pos_move.write({
                         'price_unit': float_round(new_total_value / pos_move.product_qty, precision_digits=price_unit_prec) if pos_move.product_qty else 0,
@@ -1289,7 +1289,7 @@ Please change the quantity done or the rounding precision in your settings.""",
                     })
                     merged_moves |= pos_move
                     moves_to_unlink |= neg_move
-                    if pos_move.product_uom.is_zero(pos_move.product_uom_qty):
+                    if pos_move.uom_id.is_zero(pos_move.product_uom_qty):
                         moves_to_cancel |= pos_move
                     break
                 neg_move.product_uom_qty += pos_move.product_uom_qty
@@ -1354,7 +1354,7 @@ Please change the quantity done or the rounding precision in your settings.""",
     @api.onchange('lot_ids')
     def _onchange_lot_ids(self):
         quantity = sum(ml.quantity_product_uom for ml in self.move_line_ids.filtered(lambda ml: not ml.lot_id and ml.lot_name))
-        quantity += self.product_id.uom_id._compute_quantity(len(self.lot_ids), self.product_uom)
+        quantity += self.product_id.uom_id._compute_quantity(len(self.lot_ids), self.uom_id)
         self.update({'quantity': quantity})
         base_location = self.location_id
         quants = self.env['stock.quant'].sudo().search([
@@ -1421,7 +1421,7 @@ Please change the quantity done or the rounding precision in your settings.""",
             else:
                 # Don't create picking for negative moves since they will be
                 # reverse and assign to another picking
-                moves = moves.filtered(lambda m: m.product_uom.compare(m.product_uom_qty, 0.0) >= 0)
+                moves = moves.filtered(lambda m: m.uom_id.compare(m.product_uom_qty, 0.0) >= 0)
                 if not moves:
                     continue
                 new_picking = True
@@ -1469,7 +1469,7 @@ Please change the quantity done or the rounding precision in your settings.""",
             'picking_id': self.picking_id.id,
             'location_id': self.location_id.id,
             'product_id': self.product_id.id,
-            'product_uom_id': self.product_id.uom_id.id,
+            'uom_id': self.product_id.uom_id.id,
         }
         # Select the right move lines depending of the picking type's configuration.
         move_lines = self.move_line_ids.filtered(lambda ml: not ml.lot_id and not ml.lot_name)
@@ -1573,7 +1573,7 @@ Please change the quantity done or the rounding precision in your settings.""",
             values = move._prepare_procurement_values()
             origin = move._prepare_procurement_origin()
             procurement_requests.append(self.env['stock.rule'].Procurement(
-                move.product_id, quantity, move.product_uom,
+                move.product_id, quantity, move.uom_id,
                 move.location_id, move.rule_id and move.rule_id.name or "/",
                 origin, move.company_id, values))
         self.env['stock.rule'].run(procurement_requests, raise_user_error=not self.env.context.get('from_orderpoint'))
@@ -1594,7 +1594,7 @@ Please change the quantity done or the rounding precision in your settings.""",
         if merge:
             moves = self._merge_moves(merge_into=merge_into)
 
-        neg_r_moves = moves.filtered(lambda move: move.product_uom.compare(move.product_uom_qty, 0) < 0)
+        neg_r_moves = moves.filtered(lambda move: move.uom_id.compare(move.product_uom_qty, 0) < 0)
 
         # Push remaining quantities to next step
         neg_to_push = neg_r_moves.filtered(lambda move: move.location_final_id and move.location_dest_id != move.location_final_id)
@@ -1608,7 +1608,7 @@ Please change the quantity done or the rounding precision in your settings.""",
             orig_move_ids, dest_move_ids = [], []
             for m in move.move_orig_ids | move.move_dest_ids:
                 from_loc, to_loc = m.location_id, m.location_dest_id
-                if m.product_uom.compare(m.product_uom_qty, 0) < 0:
+                if m.uom_id.compare(m.product_uom_qty, 0) < 0:
                     from_loc, to_loc = to_loc, from_loc
                 if to_loc == move.location_id:
                     orig_move_ids += m.ids
@@ -1625,7 +1625,7 @@ Please change the quantity done or the rounding precision in your settings.""",
         # Call `_action_assign` on confirmed moves eligible for auto-assignment at confirmation.
         moves._filtered_for_assign()._action_assign()
         if new_push_moves:
-            neg_push_moves = new_push_moves.filtered(lambda sm: sm.product_uom.compare(sm.product_uom_qty, 0) < 0)
+            neg_push_moves = new_push_moves.filtered(lambda sm: sm.uom_id.compare(sm.product_uom_qty, 0) < 0)
             (new_push_moves - neg_push_moves).sudo()._action_confirm()
             # Negative moves do not have any picking, so we should try to merge it with their siblings
             neg_push_moves._action_confirm(merge_into=neg_push_moves.move_orig_ids.move_dest_ids)
@@ -1662,7 +1662,7 @@ Please change the quantity done or the rounding precision in your settings.""",
 
             free_qty = max(forecasted_qties_by_loc[move.location_id][move.product_id.id], 0)
             quantity = max(move.product_qty - free_qty, 0)
-            product_uom_qty = move.product_id.uom_id._compute_quantity(quantity, move.product_uom, rounding_method='HALF-UP')
+            product_uom_qty = move.product_id.uom_id._compute_quantity(quantity, move.uom_id, rounding_method='HALF-UP')
             quantities.append(product_uom_qty)
             forecasted_qties_by_loc[move.location_id][move.product_id.id] -= min(move.product_qty, free_qty)
 
@@ -1715,7 +1715,7 @@ Please change the quantity done or the rounding precision in your settings.""",
         vals = {
             'move_id': self.id,
             'product_id': self.product_id.id,
-            'product_uom_id': self.product_uom.id,
+            'uom_id': self.uom_id.id,
             'location_id': self.location_id.id,
             'location_dest_id': self.location_dest_id.id,
             'picking_id': self.picking_id.id,
@@ -1724,13 +1724,13 @@ Please change the quantity done or the rounding precision in your settings.""",
         if quantity:
             # TODO could be also move in create/write
             rounding = self.env['decimal.precision'].precision_get('Product Unit')
-            uom_quantity = self.product_id.uom_id._compute_quantity(quantity, self.product_uom, rounding_method='HALF-UP')
+            uom_quantity = self.product_id.uom_id._compute_quantity(quantity, self.uom_id, rounding_method='HALF-UP')
             uom_quantity = float_round(uom_quantity, precision_digits=rounding)
-            uom_quantity_back_to_product_uom = self.product_uom._compute_quantity(uom_quantity, self.product_id.uom_id, rounding_method='HALF-UP')
+            uom_quantity_back_to_product_uom = self.uom_id._compute_quantity(uom_quantity, self.product_id.uom_id, rounding_method='HALF-UP')
             if float_compare(quantity, uom_quantity_back_to_product_uom, precision_digits=rounding) == 0:
                 vals = dict(vals, quantity=uom_quantity)
             else:
-                vals = dict(vals, quantity=quantity, product_uom_id=self.product_id.uom_id.id)
+                vals = dict(vals, quantity=quantity, uom_id=self.product_id.uom_id.id)
         package = None
         if reserved_quant:
             package = reserved_quant.package_id
@@ -1765,7 +1765,7 @@ Please change the quantity done or the rounding precision in your settings.""",
             owner_id = self.env['res.partner']
 
         quants = self.env['stock.quant'].with_context(packaging_uom_id=self.packaging_uom_id)._get_reserve_quantity(
-            self.product_id, location_id, need, uom_id=self.product_uom,
+            self.product_id, location_id, need, uom_id=self.uom_id,
             lot_id=lot_id, package_id=package_id, owner_id=owner_id, strict=strict)
 
         taken_quantity = 0
@@ -1788,9 +1788,9 @@ Please change the quantity done or the rounding precision in your settings.""",
             taken_quantity += quantity
             to_update = candidate_lines.get((reserved_quant.location_id, reserved_quant.lot_id, reserved_quant.package_id, reserved_quant.owner_id))
             if to_update:
-                uom_quantity = self.product_id.uom_id._compute_quantity(quantity, to_update.product_uom_id, rounding_method='HALF-UP')
+                uom_quantity = self.product_id.uom_id._compute_quantity(quantity, to_update.uom_id, rounding_method='HALF-UP')
                 uom_quantity = float_round(uom_quantity, precision_digits=rounding)
-                uom_quantity_back_to_product_uom = to_update.product_uom_id._compute_quantity(uom_quantity, self.product_id.uom_id, rounding_method='HALF-UP')
+                uom_quantity_back_to_product_uom = to_update.uom_id._compute_quantity(uom_quantity, self.product_id.uom_id, rounding_method='HALF-UP')
             if to_update and float_compare(quantity, uom_quantity_back_to_product_uom, precision_digits=rounding) == 0:
                 to_update.with_context(reserved_quant=reserved_quant).quantity += uom_quantity
             else:
@@ -1823,7 +1823,7 @@ Please change the quantity done or the rounding precision in your settings.""",
             for ml in self.move_line_ids:
                 if not ml.picked:
                     continue
-                picked_qty += ml.product_uom_id._compute_quantity(ml.quantity, self.product_uom, round=False)
+                picked_qty += ml.uom_id._compute_quantity(ml.quantity, self.uom_id, round=False)
             return picked_qty
         else:
             return self.quantity
@@ -1845,7 +1845,7 @@ Please change the quantity done or the rounding precision in your settings.""",
         for k, g in groupby(move_lines_in, key=_keys_in_groupby):
             quantity = 0
             for ml in g:
-                quantity += ml.product_uom_id._compute_quantity(ml.quantity, ml.product_id.uom_id)
+                quantity += ml.uom_id._compute_quantity(ml.quantity, ml.product_id.uom_id)
             grouped_move_lines_in[k] = quantity
 
         return grouped_move_lines_in
@@ -1869,7 +1869,7 @@ Please change the quantity done or the rounding precision in your settings.""",
         for k, g in groupby(move_lines_out_done, key=_keys_out_groupby):
             quantity = 0
             for ml in g:
-                quantity += ml.product_uom_id._compute_quantity(ml.quantity, ml.product_id.uom_id)
+                quantity += ml.uom_id._compute_quantity(ml.quantity, ml.product_id.uom_id)
             grouped_move_lines_out[k] = quantity
         for k, g in groupby(move_lines_out_reserved, key=_keys_out_groupby):
             grouped_move_lines_out[k] = sum(self.env['stock.move.line'].concat(*list(g)).mapped('quantity_product_uom'))
@@ -1919,7 +1919,7 @@ Please change the quantity done or the rounding precision in your settings.""",
             if float_compare(missing_reserved_uom_quantity, 0, precision_rounding=rounding) <= 0:
                 assigned_moves_ids.add(move.id)
                 continue
-            missing_reserved_quantity = move.product_uom._compute_quantity(missing_reserved_uom_quantity, move.product_id.uom_id, rounding_method='HALF-UP')
+            missing_reserved_quantity = move.uom_id._compute_quantity(missing_reserved_uom_quantity, move.product_id.uom_id, rounding_method='HALF-UP')
             if move._should_bypass_reservation():
                 # create the move line(s) but do not impact quants
                 if move.move_orig_ids:
@@ -1943,7 +1943,7 @@ Please change the quantity done or the rounding precision in your settings.""",
                     for _i in range(int(missing_reserved_quantity)):
                         move_line_vals_list.append(move._prepare_move_line_vals(quantity=1))
                 elif missing_reserved_quantity:
-                    to_update = move.move_line_ids.filtered(lambda ml: ml.product_uom_id == move.product_uom and
+                    to_update = move.move_line_ids.filtered(lambda ml: ml.uom_id == move.uom_id and
                                                             ml.location_id == move.location_id and
                                                             ml.location_dest_id == move.location_dest_id and
                                                             ml.picking_id == move.picking_id and
@@ -1954,13 +1954,13 @@ Please change the quantity done or the rounding precision in your settings.""",
                                                             not ml.owner_id)
                     if to_update:
                         to_update[0].quantity += move.product_id.uom_id._compute_quantity(
-                            missing_reserved_quantity, move.product_uom, rounding_method='HALF-UP')
+                            missing_reserved_quantity, move.uom_id, rounding_method='HALF-UP')
                     else:
                         move_line_vals_list.append(move._prepare_move_line_vals(quantity=missing_reserved_quantity))
                 assigned_moves_ids.add(move.id)
                 moves_to_redirect.add(move.id)
             else:
-                if move.product_uom.is_zero(move.product_uom_qty) and not force_qty:
+                if move.uom_id.is_zero(move.product_uom_qty) and not force_qty:
                     assigned_moves_ids.add(move.id)
                 elif not move.move_orig_ids:
                     if move.procure_method == 'make_to_order':
@@ -1982,7 +1982,7 @@ Please change the quantity done or the rounding precision in your settings.""",
                 else:
                     # Check what our parents brought and what our siblings took in order to
                     # determine what we can distribute.
-                    # `quantity` is in `ml.product_uom_id` and, as we will later increase
+                    # `quantity` is in `ml.uom_id` and, as we will later increase
                     # the reserved quantity on the quants, convert it here in
                     # `product_id.uom_id` (the UOM of the quants is the UOM of the product).
                     available_move_lines = move._get_available_move_lines(assigned_moves_ids, partially_available_moves_ids)
@@ -2092,7 +2092,7 @@ Please change the quantity done or the rounding precision in your settings.""",
                 # where non-scanned mls = not picked => we definitely don't want to validate them
                 ml_ids_to_unlink |= move.move_line_ids.filtered(lambda ml: not ml.picked).ids
             if (move.quantity <= 0 or not move.picked) and not move.is_inventory:
-                if move.product_uom.compare(move.product_uom_qty, 0.0) == 0 or cancel_backorder:
+                if move.uom_id.compare(move.product_uom_qty, 0.0) == 0 or cancel_backorder:
                     move._action_cancel()
         self.env['stock.move.line'].browse(ml_ids_to_unlink).unlink()
 
@@ -2109,7 +2109,7 @@ Please change the quantity done or the rounding precision in your settings.""",
         for result_package in moves_todo\
                 .move_line_ids.filtered(lambda ml: ml.picked).mapped('result_package_id')\
                 .filtered(lambda p: p.quant_ids and len(p.quant_ids) > 1):
-            if len(result_package.quant_ids.filtered(lambda q: q.product_uom_id.compare(q.quantity, 0.0) > 0).mapped('location_id')) > 1:
+            if len(result_package.quant_ids.filtered(lambda q: q.uom_id.compare(q.quantity, 0.0) > 0).mapped('location_id')) > 1:
                 raise UserError(_('You cannot move the same package content more than once in the same transfer or split the same package into two location.'))
         if any(ml.package_id and ml.package_id == ml.result_package_id for ml in moves_todo.move_line_ids):
             self.env['stock.quant']._unlink_zero_quants()
@@ -2155,7 +2155,7 @@ Please change the quantity done or the rounding precision in your settings.""",
             rounding = self.env['decimal.precision'].precision_get('Product Unit')
             if float_compare(move.quantity, move.product_uom_qty, precision_digits=rounding) < 0:
                 # Need to do some kind of conversion here
-                qty_split = move.product_uom._compute_quantity(move.product_uom_qty - move.quantity, move.product_id.uom_id, rounding_method='HALF-UP')
+                qty_split = move.uom_id._compute_quantity(move.product_uom_qty - move.quantity, move.product_id.uom_id, rounding_method='HALF-UP')
                 new_move_vals = move._split(qty_split)
                 backorder_moves_vals += new_move_vals
         backorder_moves = self.env['stock.move'].create(backorder_moves_vals)
@@ -2185,7 +2185,7 @@ Please change the quantity done or the rounding precision in your settings.""",
             'date_deadline': self.date_deadline,
         }
         if self.env.context.get('force_split_uom_id'):
-            vals['product_uom'] = self.env.context['force_split_uom_id']
+            vals['uom_id'] = self.env.context['force_split_uom_id']
         return vals
 
     def _split(self, qty, restrict_partner_id=False):
@@ -2211,8 +2211,8 @@ Please change the quantity done or the rounding precision in your settings.""",
         # quants UOM. If we're able to convert back and forth this quantity in the move's and the
         # quants UOM, the backordered move can keep the UOM of the move. Else, we'll create is in
         # the UOM of the quants.
-        uom_qty = self.product_id.uom_id._compute_quantity(qty, self.product_uom, rounding_method='HALF-UP')
-        if float_compare(qty, self.product_uom._compute_quantity(uom_qty, self.product_id.uom_id, rounding_method='HALF-UP'), precision_digits=decimal_precision) == 0:
+        uom_qty = self.product_id.uom_id._compute_quantity(qty, self.uom_id, rounding_method='HALF-UP')
+        if float_compare(qty, self.uom_id._compute_quantity(uom_qty, self.product_id.uom_id, rounding_method='HALF-UP'), precision_digits=decimal_precision) == 0:
             defaults = self._prepare_move_split_vals(uom_qty)
         else:
             defaults = self.with_context(force_split_uom_id=self.product_id.uom_id.id)._prepare_move_split_vals(qty)
@@ -2228,7 +2228,7 @@ Please change the quantity done or the rounding precision in your settings.""",
         # Update the original `product_qty` of the move. Use the general product's decimal
         # precision and not the move's UOM to handle case where the `quantity_done` is not
         # compatible with the move's UOM.
-        new_product_qty = self.product_id.uom_id._compute_quantity(max(0, self.product_qty - qty), self.product_uom, round=False)
+        new_product_qty = self.product_id.uom_id._compute_quantity(max(0, self.product_qty - qty), self.uom_id, round=False)
         new_product_qty = float_round(new_product_qty, precision_digits=self.env['decimal.precision'].precision_get('Product Unit'))
         self.with_context(do_not_unreserve=True).write({'product_uom_qty': new_product_qty})
         self._recompute_state()
@@ -2245,7 +2245,7 @@ Please change the quantity done or the rounding precision in your settings.""",
             return
         moves_state_to_write = defaultdict(set)
         for move in self:
-            rounding = move.product_uom.rounding
+            rounding = move.uom_id.rounding
             if move.state in ('cancel', 'done') or (move.state == 'draft' and not move.quantity):
                 continue
             elif float_compare(move.quantity, move.product_uom_qty, precision_rounding=rounding) >= 0:
@@ -2253,7 +2253,7 @@ Please change the quantity done or the rounding precision in your settings.""",
             elif move.quantity and float_compare(move.quantity, move.product_uom_qty, precision_rounding=rounding) <= 0:
                 moves_state_to_write['partially_available'].add(move.id)
             elif (move.procure_method == 'make_to_order' and not move.move_orig_ids) or\
-                 (move.move_orig_ids and any(orig.product_uom.compare(orig.product_uom_qty, 0) > 0
+                 (move.move_orig_ids and any(orig.uom_id.compare(orig.product_uom_qty, 0) > 0
                                              and orig.state not in ('done', 'cancel') for orig in move.move_orig_ids)):
                 # In the process of merging a negative move, we may still have a negative move in the move_orig_ids at that point.
                 moves_state_to_write['waiting'].add(move.id)
@@ -2292,28 +2292,28 @@ Please change the quantity done or the rounding precision in your settings.""",
 
     def _set_quantity_done_prepare_vals(self, qty):
         def _move_qty(qty):
-            return self.product_id.uom_id._compute_quantity(qty, self.product_uom, round=False)
+            return self.product_id.uom_id._compute_quantity(qty, self.uom_id, round=False)
 
         self.ensure_one()
         res = []
-        qty = self.product_uom._compute_quantity(qty, self.product_id.uom_id, round=False)
+        qty = self.uom_id._compute_quantity(qty, self.product_id.uom_id, round=False)
         total_qty = qty
         consumed_quant = set()
         for ml in self.move_line_ids:
             ml_qty = ml.quantity
-            if ml.product_uom_id.compare(ml_qty, 0) < 0:
+            if ml.uom_id.compare(ml_qty, 0) < 0:
                 continue
 
-            if ml.product_uom_id != self.product_id.uom_id:
-                ml_qty = ml.product_uom_id._compute_quantity(ml_qty, self.product_id.uom_id, round=False)
+            if ml.uom_id != self.product_id.uom_id:
+                ml_qty = ml.uom_id._compute_quantity(ml_qty, self.product_id.uom_id, round=False)
 
-            if self.product_uom.is_zero(_move_qty(qty)):
+            if self.uom_id.is_zero(_move_qty(qty)):
                 res.append(Command.delete(ml.id))
                 continue
 
             if ml.product_id.uom_id.compare(ml_qty, qty) > 0:
-                if ml.product_uom_id != self.product_id.uom_id:
-                    qty = ml.product_id.uom_id._compute_quantity(qty, ml.product_uom_id, round=False)
+                if ml.uom_id != self.product_id.uom_id:
+                    qty = ml.product_id.uom_id._compute_quantity(qty, ml.uom_id, round=False)
                 res.append(Command.update(ml.id, {'quantity': qty}))
                 qty = 0
                 continue
@@ -2324,7 +2324,7 @@ Please change the quantity done or the rounding precision in your settings.""",
             # remove what already on the line
             taken_qty = min(qty, ml_qty)
             qty -= taken_qty
-            if self.product_uom.compare(_move_qty(qty), 0) <= 0:
+            if self.uom_id.compare(_move_qty(qty), 0) <= 0:
                 continue
 
             # find a quant similar to the move line on which we can reserve
@@ -2338,15 +2338,15 @@ Please change the quantity done or the rounding precision in your settings.""",
             avail_qty = sum(q[1] for q in ml_quants)
             # the quant did not add the quantity reserved on this specific move line
             consumed_quant |= {q[0].id for q in ml_quants}
-            if self.product_uom.compare(avail_qty, qty) <= 0:
+            if self.uom_id.compare(avail_qty, qty) <= 0:
                 qty -= avail_qty  # decrease the target quantity for the next move lines
                 avail_qty += ml_qty  # add the actual move line quantity as we will update it and not `+=` it
-                if ml.product_uom_id != self.product_id.uom_id:
-                    avail_qty = ml.product_id.uom_id._compute_quantity(avail_qty, ml.product_uom_id, round=False)
+                if ml.uom_id != self.product_id.uom_id:
+                    avail_qty = ml.product_id.uom_id._compute_quantity(avail_qty, ml.uom_id, round=False)
                 res.append(Command.update(ml.id, {'quantity': avail_qty}))
 
         # First reserve on quants
-        if self.product_uom.compare(_move_qty(qty), 0.0) > 0:
+        if self.uom_id.compare(_move_qty(qty), 0.0) > 0:
             quants = self.env['stock.quant']._get_reserve_quantity(self.product_id, self.location_id, total_qty)
             for quant, avail_qty in quants:
                 if quant.id in consumed_quant:
@@ -2359,7 +2359,7 @@ Please change the quantity done or the rounding precision in your settings.""",
                     break
 
         # If quant is not enough, create a(some) move lines from the move itself
-        if self.product_uom.compare(_move_qty(qty), 0.0) > 0:
+        if self.uom_id.compare(_move_qty(qty), 0.0) > 0:
             if self.product_id.tracking != 'serial':
                 vals = self._prepare_move_line_vals(quantity=qty)
                 res.append((0, 0, vals))
@@ -2367,7 +2367,7 @@ Please change the quantity done or the rounding precision in your settings.""",
                 for _i in range(0, int(qty)):
                     vals = self._prepare_move_line_vals(quantity=0)
                     vals['quantity'] = 1
-                    vals['product_uom_id'] = self.product_id.uom_id.id
+                    vals['uom_id'] = self.product_id.uom_id.id
                     res.append((0, 0, vals))
         return res
 
@@ -2376,7 +2376,7 @@ Please change the quantity done or the rounding precision in your settings.""",
         Set the given quantity as quantity done on the move through the move lines. The method is
         able to handle move lines with a different UoM than the move (but honestly, this would be
         looking for trouble...).
-        @param qty: quantity in the UoM of move.product_uom
+        @param qty: quantity in the UoM of move.uom_id
         """
         existing_smls = self.move_line_ids
         self.move_line_ids = self._set_quantity_done_prepare_vals(qty)
@@ -2614,14 +2614,14 @@ Please change the quantity done or the rounding precision in your settings.""",
             **parent_record._get_product_price_and_data(self.product_id),
             'quantity': sum(
                 self.mapped(
-                    lambda line: line.product_uom._compute_quantity(
+                    lambda line: line.uom_id._compute_quantity(
                         qty=line.product_qty,
-                        to_unit=line.product_uom,
+                        to_unit=line.uom_id,
                     ),
                 ),
             ),
             'readOnly': len(self) > 1,
-            'uomDisplayName': len(self) == 1 and self.product_uom.display_name or self.product_id.uom_id.display_name,
+            'uomDisplayName': len(self) == 1 and self.uom_id.display_name or self.product_id.uom_id.display_name,
         }
 
     def _visible_quantity(self):
@@ -2653,7 +2653,7 @@ Please change the quantity done or the rounding precision in your settings.""",
             available_qty = product.qty_available
         else:
             available_qty = sum(product.with_context(lot_id=lot.id).qty_available for lot in self.lot_ids)
-        scrap_qty = self.product_uom._compute_quantity(self.quantity, self.product_id.uom_id)
+        scrap_qty = self.uom_id._compute_quantity(self.quantity, self.product_id.uom_id)
         return float_compare(available_qty, scrap_qty, precision_rounding=self.product_id.uom_id.rounding) >= 0
 
     def _action_replenish(self, values=False):
@@ -2662,7 +2662,7 @@ Please change the quantity done or the rounding precision in your settings.""",
         self.with_context(clean_context(self.env.context)).env['stock.rule'].run([self.env['stock.rule'].Procurement(
             self.product_id,
             self.quantity,
-            self.product_uom,
+            self.uom_id,
             self.location_id,
             self.origin,
             self.origin,
@@ -2689,7 +2689,7 @@ Please change the quantity done or the rounding precision in your settings.""",
 
     def action_scrap(self):
         self.ensure_one()
-        if self.product_uom.is_zero(self.quantity):
+        if self.uom_id.is_zero(self.quantity):
             raise UserError(_("You can only enter positive quantities."))
         if self.has_tracking != 'none' and not self.lot_ids:
             raise UserError(_("You must select a lot/serial number."))

--- a/addons/stock/models/stock_move_line.py
+++ b/addons/stock/models/stock_move_line.py
@@ -29,9 +29,9 @@ class StockMoveLine(models.Model):
     company_id = fields.Many2one('res.company', string='Company', readonly=True, required=True, index=True)
     product_id = fields.Many2one('product.product', 'Product', ondelete="cascade", check_company=True, domain="[('type', '!=', 'service')]", index=True)
     allowed_uom_ids = fields.Many2many('uom.uom', compute='_compute_allowed_uom_ids')
-    product_uom_id = fields.Many2one(
+    uom_id = fields.Many2one(
         'uom.uom', 'Unit', required=True, domain="[('id', 'in', allowed_uom_ids)]",
-        compute="_compute_product_uom_id", store=True, readonly=False, precompute=True,
+        compute="_compute_uom_id", store=True, readonly=False, precompute=True,
     )
     product_category_name = fields.Char(related="product_id.categ_id.complete_name", string="Product Category")
     quantity = fields.Float(
@@ -97,19 +97,19 @@ class StockMoveLine(models.Model):
     _free_reservation_index = models.Index("""(id, company_id, product_id, lot_id, location_id, owner_id, package_id)
         WHERE (state IS NULL OR state NOT IN ('cancel', 'done')) AND quantity_product_uom > 0 AND picked IS NOT TRUE""")
 
-    @api.depends('product_id', 'product_id.uom_id', 'product_id.uom_ids', 'product_id.seller_ids', 'product_id.seller_ids.product_uom_id')
+    @api.depends('product_id', 'product_id.uom_id', 'product_id.uom_ids', 'product_id.seller_ids', 'product_id.seller_ids.uom_id')
     def _compute_allowed_uom_ids(self):
         for line in self:
-            line.allowed_uom_ids = line.product_id.uom_id | line.product_id.uom_ids | line.sudo().product_id.seller_ids.product_uom_id
+            line.allowed_uom_ids = line.product_id.uom_id | line.product_id.uom_ids | line.sudo().product_id.seller_ids.uom_id
 
-    @api.depends('move_id.product_uom', 'product_id.uom_id')
-    def _compute_product_uom_id(self):
+    @api.depends('move_id.uom_id', 'product_id.uom_id')
+    def _compute_uom_id(self):
         for line in self:
-            if not line.product_uom_id:
-                if line.move_id.product_uom:
-                    line.product_uom_id = line.move_id.product_uom.id
+            if not line.uom_id:
+                if line.move_id.uom_id:
+                    line.uom_id = line.move_id.uom_id.id
                 else:
-                    line.product_uom_id = line.product_id.uom_id.id
+                    line.uom_id = line.product_id.uom_id.id
 
     @api.depends('picking_id.picking_type_id', 'product_id.tracking')
     def _compute_lots_visible(self):
@@ -151,23 +151,23 @@ class StockMoveLine(models.Model):
         for record in self:
             if not record.quant_id or record.quantity:
                 continue
-            product_uom = record.product_id.uom_id
-            sml_uom = record.product_uom_id
+            uom_id = record.product_id.uom_id
+            sml_uom = record.uom_id
             move_visible_quantity = record.move_id and record.move_id._visible_quantity() or 0.0
 
-            move_demand = record.move_id.product_uom._compute_quantity(record.move_id.product_uom_qty, sml_uom, rounding_method='HALF-UP')
-            move_quantity = record.move_id.product_uom._compute_quantity(move_visible_quantity, sml_uom, rounding_method='HALF-UP')
-            quant_qty = product_uom._compute_quantity(record.quant_id.available_quantity, sml_uom, rounding_method='HALF-UP')
+            move_demand = record.move_id.uom_id._compute_quantity(record.move_id.product_uom_qty, sml_uom, rounding_method='HALF-UP')
+            move_quantity = record.move_id.uom_id._compute_quantity(move_visible_quantity, sml_uom, rounding_method='HALF-UP')
+            quant_qty = uom_id._compute_quantity(record.quant_id.available_quantity, sml_uom, rounding_method='HALF-UP')
 
             if sml_uom.compare(move_demand, move_quantity) > 0:
                 record.quantity = max(0, min(quant_qty, move_demand - move_quantity))
             else:
                 record.quantity = max(0, quant_qty)
 
-    @api.depends('quantity', 'product_uom_id')
+    @api.depends('quantity', 'uom_id')
     def _compute_quantity_product_uom(self):
         for line in self:
-            line.quantity_product_uom = line.product_uom_id._compute_quantity(line.quantity, line.product_id.uom_id, rounding_method='HALF-UP')
+            line.quantity_product_uom = line.uom_id._compute_quantity(line.quantity, line.product_id.uom_id, rounding_method='HALF-UP')
 
     @api.constrains('lot_id', 'product_id')
     def _check_lot_product(self):
@@ -184,7 +184,7 @@ class StockMoveLine(models.Model):
         if any(ml.quantity < 0 for ml in self):
             raise ValidationError(_('You can not enter negative quantities.'))
 
-    @api.onchange('product_id', 'product_uom_id')
+    @api.onchange('product_id', 'uom_id')
     def _onchange_product_id(self):
         if self.product_id:
             self.lots_visible = self.product_id.tracking != 'none'
@@ -234,7 +234,7 @@ class StockMoveLine(models.Model):
                 res['warning'] = {'title': _('Warning'), 'message': message}
         return res
 
-    @api.onchange('quantity', 'product_uom_id')
+    @api.onchange('quantity', 'uom_id')
     def _onchange_quantity(self):
         """ When the user is encoding a move line for a tracked product, we apply some logic to
         help him. This onchange will warn him if he set `quantity` to a non-supported value.
@@ -245,7 +245,7 @@ class StockMoveLine(models.Model):
                 raise UserError(_('You can only process 1.0 %s of products with unique serial number.', self.product_id.uom_id.name))
         return res
 
-    @api.onchange('result_package_id', 'product_id', 'product_uom_id', 'quantity')
+    @api.onchange('result_package_id', 'product_id', 'uom_id', 'quantity')
     def _onchange_putaway_location(self):
         default_dest_location = self._get_default_dest_location()
         if not self.id and self.env.user.has_group('stock.group_stock_multi_locations') and self.product_id and self.quantity_product_uom \
@@ -295,7 +295,7 @@ class StockMoveLine(models.Model):
     def _get_putaway_additional_qty(self):
         addtional_qty = {}
         for ml in self._origin:
-            qty = ml.product_uom_id._compute_quantity(ml.quantity, ml.product_id.uom_id)
+            qty = ml.uom_id._compute_quantity(ml.quantity, ml.product_id.uom_id)
             addtional_qty[ml.location_dest_id.id] = addtional_qty.get(ml.location_dest_id.id, 0) - qty
         return addtional_qty
 
@@ -401,7 +401,7 @@ class StockMoveLine(models.Model):
             if ml.state == 'done':
                 if ml.product_id.is_storable:
                     Quant = self.env['stock.quant']
-                    quantity = ml.product_uom_id._compute_quantity(ml.quantity, ml.move_id.product_id.uom_id, rounding_method='HALF-UP')
+                    quantity = ml.uom_id._compute_quantity(ml.quantity, ml.move_id.product_id.uom_id, rounding_method='HALF-UP')
                     available_qty, in_date = Quant._update_available_quantity(ml.product_id, ml.location_id, -quantity, lot_id=ml.lot_id, package_id=ml.package_id, owner_id=ml.owner_id)
                     if available_qty < 0 and ml.lot_id:
                         # see if we can compensate the negative quants with some untracked quants
@@ -438,7 +438,7 @@ class StockMoveLine(models.Model):
             ('package_id', 'stock.package'),
             ('result_package_id', 'stock.package'),
             ('owner_id', 'res.partner'),
-            ('product_uom_id', 'uom.uom')
+            ('uom_id', 'uom.uom')
         ]
         if vals.get('quant_id'):
             vals.update(self._copy_quant_info(vals))
@@ -459,8 +459,8 @@ class StockMoveLine(models.Model):
             for ml in self:
                 if not ml.product_id.is_storable or ml.state == 'done':
                     continue
-                if 'quantity' in vals or 'product_uom_id' in vals:
-                    new_ml_uom = updates.get('product_uom_id', ml.product_uom_id)
+                if 'quantity' in vals or 'uom_id' in vals:
+                    new_ml_uom = updates.get('uom_id', ml.uom_id)
                     new_reserved_qty = new_ml_uom._compute_quantity(
                         vals.get('quantity', ml.quantity), ml.product_id.uom_id, rounding_method='HALF-UP')
                     # Make sure `reserved_uom_qty` is not negative.
@@ -470,7 +470,7 @@ class StockMoveLine(models.Model):
                     new_reserved_qty = ml.quantity_product_uom
 
                 # Unreserve the old charateristics of the move line.
-                if not ml.product_uom_id.is_zero(ml.quantity_product_uom):
+                if not ml.uom_id.is_zero(ml.quantity_product_uom):
                     ml._synchronize_quant(-ml.quantity_product_uom, ml.location_id, action="reserved")
 
                 # Reserve the maximum available of the new charateristics of the move line.
@@ -480,7 +480,7 @@ class StockMoveLine(models.Model):
                         lot=updates.get('lot_id', ml.lot_id), package=updates.get('package_id', ml.package_id),
                         owner=updates.get('owner_id', ml.owner_id))
 
-                if ('quantity' in vals and vals['quantity'] != ml.quantity) or 'product_uom_id' in vals:
+                if ('quantity' in vals and vals['quantity'] != ml.quantity) or 'uom_id' in vals:
                     moves_to_recompute_state |= ml.move_id
 
         # When editing a done move line, the reserved availability of a potential chained move is impacted. Take care of running again `_action_assign` on the concerned moves.
@@ -489,7 +489,7 @@ class StockMoveLine(models.Model):
             next_moves = self.env['stock.move']
             mls = self.filtered(lambda ml: ml.move_id.state == 'done' and ml.product_id.is_storable)
             if not updates:  # we can skip those where quantity is already good up to UoM rounding
-                mls = mls.filtered(lambda ml: not ml.product_uom_id.is_zero(ml.quantity - vals['quantity']))
+                mls = mls.filtered(lambda ml: not ml.uom_id.is_zero(ml.quantity - vals['quantity']))
             for ml in mls:
                 # undo the original move line
                 in_date = ml._synchronize_quant(-ml.quantity_product_uom, ml.location_dest_id, package=ml.result_package_id)[1]
@@ -506,7 +506,7 @@ class StockMoveLine(models.Model):
                 move_done._check_quantity()
 
         # update the date when it seems like (additional) quantities are "done" and the date hasn't been manually updated
-        if 'date' not in vals and ('product_uom_id' in vals or 'quantity' in vals or vals.get('picked', False)):
+        if 'date' not in vals and ('uom_id' in vals or 'quantity' in vals or vals.get('picked', False)):
             updated_ml_ids = set()
             for ml in self:
                 if ml.state in ['draft', 'cancel', 'done']:
@@ -514,10 +514,10 @@ class StockMoveLine(models.Model):
                 if vals.get('picked', False) and not ml.picked:
                     updated_ml_ids.add(ml.id)
                     continue
-                if ('quantity' in vals or 'product_uom_id' in vals) and ml.picked:
-                    new_qty = updates.get('product_uom_id', ml.product_uom_id)._compute_quantity(vals.get('quantity', ml.quantity), ml.product_id.uom_id, rounding_method='HALF-UP')
-                    old_qty = ml.product_uom_id._compute_quantity(ml.quantity, ml.product_id.uom_id, rounding_method='HALF-UP')
-                    if ml.product_uom_id.compare(old_qty, new_qty) < 0:
+                if ('quantity' in vals or 'uom_id' in vals) and ml.picked:
+                    new_qty = updates.get('uom_id', ml.uom_id)._compute_quantity(vals.get('quantity', ml.quantity), ml.product_id.uom_id, rounding_method='HALF-UP')
+                    old_qty = ml.uom_id._compute_quantity(ml.quantity, ml.product_id.uom_id, rounding_method='HALF-UP')
+                    if ml.uom_id.compare(old_qty, new_qty) < 0:
                         updated_ml_ids.add(ml.id)
             self.env['stock.move.line'].browse(updated_ml_ids).date = fields.Datetime.now()
 
@@ -606,17 +606,17 @@ class StockMoveLine(models.Model):
         ml_ids_to_check = defaultdict(OrderedSet)
 
         for ml in self:
-            # Check here if `ml.quantity` respects the rounding of `ml.product_uom_id`.
-            uom_qty = ml.product_uom_id.round(ml.quantity, rounding_method='HALF-UP')
+            # Check here if `ml.quantity` respects the rounding of `ml.uom_id`.
+            uom_qty = ml.uom_id.round(ml.quantity, rounding_method='HALF-UP')
             precision_digits = self.env['decimal.precision'].precision_get('Product Unit')
             quantity = float_round(ml.quantity, precision_digits=precision_digits, rounding_method='HALF-UP')
             if float_compare(uom_qty, quantity, precision_digits=precision_digits) != 0:
                 raise UserError(_('The quantity done for the product "%(product)s" doesn\'t respect the rounding precision '
                                   'defined on the unit of measure "%(unit)s". Please change the quantity done or the '
                                   'rounding precision of your unit of measure.',
-                                  product=ml.product_id.display_name, unit=ml.product_uom_id.name))
+                                  product=ml.product_id.display_name, unit=ml.uom_id.name))
 
-            qty_done_float_compared = ml.product_uom_id.compare(ml.quantity, 0)
+            qty_done_float_compared = ml.uom_id.compare(ml.quantity, 0)
             if qty_done_float_compared > 0:
                 if ml.product_id.tracking == 'none':
                     continue
@@ -712,7 +712,7 @@ class StockMoveLine(models.Model):
         package = quants_value.get('package', self.package_id)
         owner = quants_value.get('owner', self.owner_id)
         available_qty = 0
-        if not self.product_id.is_storable or self.product_uom_id.is_zero(quantity):
+        if not self.product_id.is_storable or self.uom_id.is_zero(quantity):
             return 0, False
         if action == "available":
             available_qty, in_date = self.env['stock.quant']._update_available_quantity(self.product_id, location, quantity, lot_id=lot, package_id=package, owner_id=owner, in_date=in_date)
@@ -831,13 +831,13 @@ class StockMoveLine(models.Model):
 
         for candidate in outdated_candidates:
             move_to_reassign |= candidate.move_id
-            if self.product_uom_id.compare(candidate.quantity_product_uom, quantity) <= 0:
+            if self.uom_id.compare(candidate.quantity_product_uom, quantity) <= 0:
                 quantity -= candidate.quantity_product_uom
                 to_unlink_candidate_ids.add(candidate.id)
-                if self.product_uom_id.is_zero(quantity):
+                if self.uom_id.is_zero(quantity):
                     break
             else:
-                candidate.quantity -= candidate.product_id.uom_id._compute_quantity(quantity, candidate.product_uom_id, rounding_method='HALF-UP')
+                candidate.quantity -= candidate.product_id.uom_id._compute_quantity(quantity, candidate.uom_id, rounding_method='HALF-UP')
                 break
 
         move_line_to_unlink = self.env['stock.move.line'].browse(to_unlink_candidate_ids)
@@ -851,7 +851,7 @@ class StockMoveLine(models.Model):
 
     def _get_aggregated_properties(self, move_line=False, move=False):
         move = move or move_line.move_id
-        uom = move.product_uom or move_line.product_uom_id
+        uom = move.uom_id or move_line.uom_id
         packaging_uom = move.packaging_uom_id
         name = move.product_id.display_name
         description = move.description_picking or ""
@@ -865,7 +865,7 @@ class StockMoveLine(models.Model):
             'line_key': line_key,
             'name': name,
             'description': description,
-            'product_uom': uom,
+            'uom_id': uom,
             'packaging_uom_id': packaging_uom,
             'move': move,
         }
@@ -883,7 +883,7 @@ class StockMoveLine(models.Model):
         the products by (i.e. so data is not lost). This function purposely ignores lots/SNs because these are
         expected to already be properly grouped by line.
 
-        returns: dictionary {product_id+name+description+uom: {product, name, description, quantity, product_uom}, ...}
+        returns: dictionary {product_id+name+description+uom: {product, name, description, quantity, uom_id}, ...}
         """
         aggregated_move_lines = {}
 
@@ -898,9 +898,9 @@ class StockMoveLine(models.Model):
             if kwargs.get('except_package') and move_line.result_package_id:
                 continue
             aggregated_properties = self._get_aggregated_properties(move_line=move_line)
-            line_key, uom = aggregated_properties['line_key'], aggregated_properties['product_uom']
-            quantity = move_line.product_uom_id._compute_quantity(move_line.quantity, uom)
-            packaging_quantity = move_line.product_uom_id._compute_quantity(quantity, move_line.move_id.packaging_uom_id)
+            line_key, uom = aggregated_properties['line_key'], aggregated_properties['uom_id']
+            quantity = move_line.uom_id._compute_quantity(move_line.quantity, uom)
+            packaging_quantity = move_line.uom_id._compute_quantity(quantity, move_line.move_id.packaging_uom_id)
             if line_key not in aggregated_move_lines:
                 qty_ordered = None
                 packaging_qty_ordered = None
@@ -916,8 +916,8 @@ class StockMoveLine(models.Model):
                     previous_move_lines = move_line.move_id.move_line_ids.filtered(
                         lambda ml: line_key.startswith(self._get_aggregated_properties(move=ml.move_id)['line_key']) and ml.id != move_line.id
                     )
-                    qty_ordered -= sum([m.product_uom_id._compute_quantity(m.quantity, uom) for m in previous_move_lines])
-                    packaging_qty_ordered = move_line.product_uom_id._compute_quantity(qty_ordered, move_line.move_id.packaging_uom_id)
+                    qty_ordered -= sum(m.uom_id._compute_quantity(m.quantity, uom) for m in previous_move_lines)
+                    packaging_qty_ordered = move_line.uom_id._compute_quantity(qty_ordered, move_line.move_id.packaging_uom_id)
                 aggregated_move_lines[line_key] = {
                     **aggregated_properties,
                     'quantity': quantity,
@@ -939,7 +939,7 @@ class StockMoveLine(models.Model):
         pickings = (self.picking_id | backorders)
         for empty_move in pickings.move_ids:
             to_bypass = False
-            if not (empty_move.product_uom_qty and empty_move.product_uom.is_zero(empty_move.quantity)):
+            if not (empty_move.product_uom_qty and empty_move.uom_id.is_zero(empty_move.quantity)):
                 continue
             if empty_move.state != "cancel":
                 if empty_move.state != "confirmed" or empty_move.move_line_ids:
@@ -998,7 +998,7 @@ class StockMoveLine(models.Model):
         return {
             'product_id': self.product_id.id,
             'product_uom_qty': 0 if self.picking_id and self.picking_id.state != 'done' else self.quantity,
-            'product_uom': self.product_uom_id.id,
+            'uom_id': self.uom_id.id,
             'location_id': self.picking_id.location_id.id,
             'location_dest_id': self.picking_id.location_dest_id.id,
             'picked': self.picked,
@@ -1155,7 +1155,7 @@ class StockMoveLine(models.Model):
         if len(self.picking_type_id) > 1:
             raise UserError(_('You cannot pack products into the same package when they are from different transfers with different operation types'))
 
-        quantity_move_lines = self.filtered(lambda ml: ml.state not in ('done', 'cancel') and ml.product_uom_id.compare(ml.quantity, 0.0) > 0)
+        quantity_move_lines = self.filtered(lambda ml: ml.state not in ('done', 'cancel') and ml.uom_id.compare(ml.quantity, 0.0) > 0)
         if picked_first:
             picked_move_lines = quantity_move_lines.filtered(lambda ml: ml.picked)
             if picked_move_lines:
@@ -1172,7 +1172,7 @@ class StockMoveLine(models.Model):
         return {
             'inventory_name': _('%s [reverted]', self.reference),
             'product_id': self.product_id.id,
-            'product_uom': self.product_uom_id.id,
+            'uom_id': self.uom_id.id,
             'product_uom_qty': self.quantity,
             'company_id': self.company_id.id or self.env.company.id,
             'state': 'confirmed',
@@ -1182,7 +1182,7 @@ class StockMoveLine(models.Model):
             'picked': True,
             'move_line_ids': [(0, 0, {
                 'product_id': self.product_id.id,
-                'product_uom_id': self.product_uom_id.id,
+                'uom_id': self.uom_id.id,
                 'quantity': self.quantity,
                 'location_id': self.location_dest_id.id,
                 'location_dest_id': self.location_id.id,
@@ -1200,7 +1200,7 @@ class StockMoveLine(models.Model):
         self = self.with_context(inventory_mode=False)
         processed_move_line = self.env['stock.move.line']
         for move_line in self:
-            if move_line.is_inventory and not move_line.product_uom_id.is_zero(move_line.quantity):
+            if move_line.is_inventory and not move_line.uom_id.is_zero(move_line.quantity):
                 processed_move_line += move_line
                 move_vals.append(move_line._get_revert_inventory_move_values())
         if not processed_move_line:

--- a/addons/stock/models/stock_orderpoint.py
+++ b/addons/stock/models/stock_orderpoint.py
@@ -51,9 +51,9 @@ class StockWarehouseOrderpoint(models.Model):
             " [('is_storable', '=', True)]"),
         ondelete='cascade', required=True, check_company=True, index=True)
     product_category_id = fields.Many2one('product.category', name='Product Category', related='product_id.categ_id')
-    product_uom = fields.Many2one(
+    uom_id = fields.Many2one(
         'uom.uom', 'Unit', related='product_id.uom_id')
-    product_uom_name = fields.Char(string='Product unit of measure label', related='product_uom.display_name', readonly=True)
+    product_uom_name = fields.Char(string='Product unit of measure label', related='uom_id.display_name', readonly=True)
     product_min_qty = fields.Float(
         'Min Quantity', digits='Product Unit', required=True, default=0.0,
         help="The minimum Stock level that will trigger a replenishment.")
@@ -212,12 +212,12 @@ class StockWarehouseOrderpoint(models.Model):
             if orderpoint.product_max_qty < orderpoint.product_min_qty or not orderpoint.product_max_qty:
                 orderpoint.product_max_qty = orderpoint.product_min_qty
 
-    @api.depends('route_id', 'product_id', 'product_id.seller_ids', 'product_id.seller_ids.product_uom_id')
+    @api.depends('route_id', 'product_id', 'product_id.seller_ids', 'product_id.seller_ids.uom_id')
     def _compute_allowed_replenishment_uom_ids(self):
         for orderpoint in self:
             orderpoint.allowed_replenishment_uom_ids = orderpoint.product_id.uom_ids
             if 'buy' in orderpoint.rule_ids.mapped('action'):
-                orderpoint.allowed_replenishment_uom_ids += orderpoint.product_id.seller_ids.product_uom_id
+                orderpoint.allowed_replenishment_uom_ids += orderpoint.product_id.seller_ids.uom_id
 
     @api.depends('allowed_replenishment_uom_ids')
     def _compute_replenishment_uom_id_placeholder(self):
@@ -282,16 +282,16 @@ class StockWarehouseOrderpoint(models.Model):
     @api.depends('product_id', 'qty_to_order', 'product_max_qty')
     def _compute_unwanted_replenish(self):
         for orderpoint in self:
-            if not orderpoint.product_id or orderpoint.product_uom.is_zero(orderpoint.qty_to_order) or orderpoint.product_uom.compare(orderpoint.product_max_qty, 0) == -1:
+            if not orderpoint.product_id or orderpoint.uom_id.is_zero(orderpoint.qty_to_order) or orderpoint.uom_id.compare(orderpoint.product_max_qty, 0) == -1:
                 orderpoint.unwanted_replenish = False
             else:
                 after_replenish_qty = orderpoint.product_id.with_context(company_id=orderpoint.company_id.id, location=orderpoint.location_id.id).virtual_available + orderpoint.qty_to_order
-                orderpoint.unwanted_replenish = orderpoint.product_uom.compare(after_replenish_qty, orderpoint.product_max_qty) > 0
+                orderpoint.unwanted_replenish = orderpoint.uom_id.compare(after_replenish_qty, orderpoint.product_max_qty) > 0
 
     @api.onchange('product_id')
     def _onchange_product_id(self):
         if self.product_id:
-            self.product_uom = self.product_id.uom_id.id
+            self.uom_id = self.product_id.uom_id.id
 
     @api.model_create_multi
     def create(self, vals_list):
@@ -417,7 +417,7 @@ class StockWarehouseOrderpoint(models.Model):
     'product_id', 'location_id', 'product_id.seller_ids.delay', 'company_id.horizon_days')
     def _compute_qty_to_order_computed(self):
         def to_compute(orderpoint):
-            rounding = orderpoint.product_uom.rounding
+            rounding = orderpoint.uom_id.rounding
             # The check is on purpose. We only want to consider the horizon days if the forecast is negative and
             # there is already something to resupply base on lead times.
             return (
@@ -471,7 +471,7 @@ class StockWarehouseOrderpoint(models.Model):
         qty_in_progress = qty_in_progress_by_orderpoint.get(self.id)
         if qty_in_progress is None:
             qty_in_progress = self._quantity_in_progress()[self.id]
-        rounding = self.product_uom.rounding
+        rounding = self.uom_id.rounding
         # The check is on purpose. We only want to consider the horizon days if the forecast is negative and
         # there is already something to resupply base on lead times.
         if float_compare(self.qty_forecast, self.product_min_qty, precision_rounding=rounding) < 0:
@@ -735,14 +735,14 @@ class StockWarehouseOrderpoint(models.Model):
                             origin = '%s - %s' % (orderpoint.display_name, ','.join(origins.mapped('name')))
                         else:
                             origin = orderpoint.name
-                        if orderpoint.product_uom.compare(orderpoint.qty_to_order, 0.0) == 1:
+                        if orderpoint.uom_id.compare(orderpoint.qty_to_order, 0.0) == 1:
                             date = orderpoint._get_orderpoint_procurement_date()
                             global_horizon_days = orderpoint.get_horizon_days()
                             if global_horizon_days:
                                 date -= relativedelta.relativedelta(days=int(global_horizon_days))
                             values = orderpoint._prepare_procurement_values(date=date)
                             procurements.append(self.env['stock.rule'].Procurement(
-                                orderpoint.product_id, orderpoint.qty_to_order, orderpoint.product_uom,
+                                orderpoint.product_id, orderpoint.qty_to_order, orderpoint.uom_id,
                                 orderpoint.location_id, orderpoint.name, origin,
                                 orderpoint.company_id, values))
 

--- a/addons/stock/models/stock_package.py
+++ b/addons/stock/models/stock_package.py
@@ -117,7 +117,7 @@ class StockPackage(models.Model):
 
         display_uom = self.env.user.has_group('uom.group_uom')
         for package in self:
-            package_content = package.contained_quant_ids.grouped(lambda q: (q.product_uom_id, q.product_id))
+            package_content = package.contained_quant_ids.grouped(lambda q: (q.uom_id, q.product_id))
             package_content = [(uom.name, product.display_name, sum(quants.mapped('quantity'))) for ((uom, product), quants) in package_content.items()]
             package.content_description = format_list(self.env, [format_content(qty, uom_name, product_name, display_uom) for (uom_name, product_name, qty) in package_content])
 
@@ -158,7 +158,7 @@ class StockPackage(models.Model):
         for package in self:
             package.location_id = False
             package.company_id = False
-            quants = package.quant_ids.filtered(lambda q: q.product_uom_id.compare(q.quantity, 0) > 0)
+            quants = package.quant_ids.filtered(lambda q: q.uom_id.compare(q.quantity, 0) > 0)
             if quants:
                 package.location_id = quants[0].location_id
                 if all(q.company_id == quants[0].company_id for q in package.quant_ids):
@@ -446,13 +446,13 @@ class StockPackage(models.Model):
 
             res_groups = self.env['stock.move.line']._read_group(
                 [('result_package_id', 'in', all_pack_ids), ('product_id', '!=', False), ('picking_id', '=', picking_id)],
-                ['result_package_id', 'product_id', 'product_uom_id', 'quantity'],
+                ['result_package_id', 'product_id', 'uom_id', 'quantity'],
                 ['__count'],
             )
-            for result_package, product, product_uom, quantity, count in res_groups:
+            for result_package, product, uom_id, quantity, count in res_groups:
                 package_weights[result_package.id] += (
                     count
-                    * product_uom._compute_quantity(quantity, product.uom_id)
+                    * uom_id._compute_quantity(quantity, product.uom_id)
                     * product.weight
                 )
         for package in self:
@@ -494,7 +494,7 @@ class StockPackage(models.Model):
             if len(new_location) > 1:
                 raise UserError(self.env._("Packages %(duplicate_names)s are moved to different locations while being in the same container %(container_name)s.",
                                             duplicate_names=packages.mapped('name'), container_name=container_package.name))
-            contained_quants = container_package.contained_quant_ids.filtered(lambda q: not float_is_zero(q.quantity, precision_rounding=q.product_uom_id.rounding))
+            contained_quants = container_package.contained_quant_ids.filtered(lambda q: not float_is_zero(q.quantity, precision_rounding=q.uom_id.rounding))
             if contained_quants and contained_quants.location_id != new_location:
                 old_location = contained_quants.location_id - new_location
                 raise UserError(self.env._("Can't move a container having packages in another location (%(old_location)s) to a different location (%(new_location)s).",

--- a/addons/stock/models/stock_picking.py
+++ b/addons/stock/models/stock_picking.py
@@ -851,18 +851,18 @@ class StockPicking(models.Model):
             else:
                 picking.scheduled_date = max(moves_dates, default=picking.scheduled_date or fields.Datetime.now())
 
-    @api.depends('move_line_ids', 'move_line_ids.result_package_id', 'move_line_ids.product_uom_id', 'move_line_ids.quantity')
+    @api.depends('move_line_ids', 'move_line_ids.result_package_id', 'move_line_ids.uom_id', 'move_line_ids.quantity')
     def _compute_bulk_weight(self):
         picking_weights = defaultdict(float)
         res_groups = self.env['stock.move.line']._read_group(
             [('picking_id', 'in', self.ids), ('product_id', '!=', False), ('result_package_id', '=', False)],
-            ['picking_id', 'product_id', 'product_uom_id', 'quantity'],
+            ['picking_id', 'product_id', 'uom_id', 'quantity'],
             ['__count'],
         )
-        for picking, product, product_uom, quantity, count in res_groups:
+        for picking, product, uom_id, quantity, count in res_groups:
             picking_weights[picking.id] += (
                 count
-                * product_uom._compute_quantity(quantity, product.uom_id)
+                * uom_id._compute_quantity(quantity, product.uom_id)
                 * product.weight
             )
         for picking in self:
@@ -890,7 +890,7 @@ class StockPicking(models.Model):
         for picking in self:
             volume = 0
             for move in picking.move_ids:
-                volume += move.product_uom._compute_quantity(move.quantity, move.product_id.uom_id) * move.product_id.volume
+                volume += move.uom_id._compute_quantity(move.quantity, move.product_id.uom_id) * move.product_id.volume
             picking.shipping_volume = volume
 
     @api.depends('move_ids.date_deadline', 'move_type')
@@ -954,7 +954,7 @@ class StockPicking(models.Model):
                 continue
             picking.show_check_availability = any(
                 move.state in ('waiting', 'confirmed', 'partially_available') and
-                move.product_uom.compare(move.product_uom_qty, 0)
+                move.uom_id.compare(move.product_uom_qty, 0)
                 for move in picking.move_ids
             )
 
@@ -1312,7 +1312,7 @@ class StockPicking(models.Model):
 
         def get_line_with_done_qty_ids(move_lines):
             # Get only move_lines that has some quantity set.
-            return move_lines.filtered(lambda ml: ml.product_id and ml.product_id.tracking != 'none' and ml.picked and ml.product_uom_id.compare(ml.quantity, 0)).ids
+            return move_lines.filtered(lambda ml: ml.product_id and ml.product_id.tracking != 'none' and ml.picked and ml.uom_id.compare(ml.quantity, 0)).ids
 
         if separate_pickings:
             # If pickings are checked independently, get full/partial move_lines depending if each picking has no quantity set.
@@ -1378,7 +1378,7 @@ class StockPicking(models.Model):
         draft_picking = self.filtered(lambda p: p.state == 'draft')
         draft_picking.action_confirm()
         for move in draft_picking.move_ids:
-            if move.product_uom.is_zero(move.quantity) and not move.product_uom.is_zero(move.product_uom_qty):
+            if move.uom_id.is_zero(move.quantity) and not move.uom_id.is_zero(move.product_uom_qty):
                 move.quantity = move.product_uom_qty
 
         # Sanity checks.
@@ -1436,11 +1436,11 @@ class StockPicking(models.Model):
         return True
 
     def action_split_transfer(self):
-        if all(m.product_uom.is_zero(m.quantity) for m in self.move_ids):
+        if all(m.uom_id.is_zero(m.quantity) for m in self.move_ids):
             raise UserError(_("%s: Nothing to split. Fill the quantities you want in a new transfer in the done quantities", self.display_name))
-        if all(m.product_uom.compare(m.quantity, m.product_uom_qty) == 0 for m in self.move_ids):
+        if all(m.uom_id.compare(m.quantity, m.product_uom_qty) == 0 for m in self.move_ids):
             raise UserError(_("%s: Nothing to split, all demand is done. For split you need at least one line not fully fulfilled", self.display_name))
-        if any(m.product_uom.compare(m.quantity, m.product_uom_qty) > 0 for m in self.move_ids):
+        if any(m.uom_id.compare(m.quantity, m.product_uom_qty) > 0 for m in self.move_ids):
             raise UserError(_("%s: Can't split: quantities done can't be above demand", self.display_name))
 
         moves = self.move_ids.filtered(lambda m: m.state not in ('done', 'cancel') and m.quantity != 0)
@@ -2137,7 +2137,7 @@ class StockPicking(models.Model):
             move_line_vals.append({
                 'product_id': package_quant.product_id.id,
                 'quantity': package_quant.quantity,
-                'product_uom_id': package_quant.product_uom_id.id,
+                'uom_id': package_quant.uom_id.id,
                 'location_id': package_quant.location_id.id,
                 'location_dest_id': self.location_dest_id.id,
                 'picking_id': self.id,

--- a/addons/stock/models/stock_quant.py
+++ b/addons/stock/models/stock_quant.py
@@ -49,7 +49,7 @@ class StockQuant(models.Model):
     product_tmpl_id = fields.Many2one(
         'product.template', string='Product Template',
         related='product_id.product_tmpl_id')
-    product_uom_id = fields.Many2one(
+    uom_id = fields.Many2one(
         'uom.uom', 'Unit',
         readonly=True, related='product_id.uom_id')
     is_favorite = fields.Boolean(related='product_tmpl_id.is_favorite')
@@ -198,14 +198,14 @@ class StockQuant(models.Model):
     def _compute_is_outdated(self):
         self.is_outdated = False
         for quant in self:
-            if quant.product_id and quant.product_uom_id.compare(quant.inventory_quantity - quant.inventory_diff_quantity, quant.quantity) and quant.inventory_quantity_set:
+            if quant.product_id and quant.uom_id.compare(quant.inventory_quantity - quant.inventory_diff_quantity, quant.quantity) and quant.inventory_quantity_set:
                 quant.is_outdated = True
 
     def _search_is_outdated(self, operator, value):
         if operator != 'in':
             return NotImplemented
         quant_ids = self.search([('inventory_quantity_set', '=', True)])
-        quant_ids = quant_ids.filtered(lambda quant: quant.product_uom_id.compare(quant.inventory_quantity - quant.inventory_diff_quantity, quant.quantity)).ids
+        quant_ids = quant_ids.filtered(lambda quant: quant.uom_id.compare(quant.inventory_quantity - quant.inventory_diff_quantity, quant.quantity)).ids
         return [('id', 'in', quant_ids)]
 
     @api.depends('quantity')
@@ -999,10 +999,10 @@ class StockQuant(models.Model):
         for quant in self:
             # if inventory applied from product's inverse_qty and the inventory_diff_quantity is 0,
             # we skip creating a move with 0 quantity.
-            if quant.env.context.get('from_inverse_qty') and quant.product_uom_id.compare(quant.inventory_diff_quantity, 0) == 0:
+            if quant.env.context.get('from_inverse_qty') and quant.uom_id.compare(quant.inventory_diff_quantity, 0) == 0:
                 continue
             # Create and validate a move so that the quant matches its `inventory_quantity`.
-            if quant.product_uom_id.compare(quant.inventory_diff_quantity, 0) > 0:
+            if quant.uom_id.compare(quant.inventory_diff_quantity, 0) > 0:
                 move_vals.append(
                     quant._get_inventory_move_values(quant.inventory_diff_quantity,
                                                      quant.product_id.with_company(quant.company_id).property_stock_inventory,
@@ -1055,7 +1055,7 @@ class StockQuant(models.Model):
             incoming_dates = []
         else:
             incoming_dates = [quant.in_date for quant in quants if quant.in_date and
-                              quant.product_uom_id.compare(quant.quantity, 0) > 0]
+                              quant.uom_id.compare(quant.quantity, 0) > 0]
         if in_date:
             incoming_dates += [in_date]
         # If multiple incoming dates are available for a given lot_id/package_id/owner_id, we
@@ -1254,7 +1254,7 @@ class StockQuant(models.Model):
 
         res = {
             'product_id': self.product_id.id,
-            'product_uom': self.product_uom_id.id,
+            'uom_id': self.uom_id.id,
             'product_uom_qty': qty,
             'company_id': self.company_id.id or self.env.company.id,
             'state': 'confirmed',
@@ -1265,7 +1265,7 @@ class StockQuant(models.Model):
             'picked': True,
             'move_line_ids': [(0, 0, {
                 'product_id': self.product_id.id,
-                'product_uom_id': self.product_uom_id.id,
+                'uom_id': self.uom_id.id,
                 'quantity': qty,
                 'location_id': location_id.id,
                 'location_dest_id': location_dest_id.id,
@@ -1356,9 +1356,9 @@ class StockQuant(models.Model):
 
         # Quantity part.
         if self.tracking != 'serial' or self.quantity > 1:
-            quantity_ai = gs1_quantity_rules_ai_by_uom.get(self.product_uom_id.id)
+            quantity_ai = gs1_quantity_rules_ai_by_uom.get(self.uom_id.id)
             if quantity_ai:
-                qty_str = str(int(self.quantity / self.product_uom_id.rounding))
+                qty_str = str(int(self.quantity / self.uom_id.rounding))
                 if len(qty_str) <= 6:
                     barcode += quantity_ai + '0' * (6 - len(qty_str)) + qty_str
             else:

--- a/addons/stock/models/stock_rule.py
+++ b/addons/stock/models/stock_rule.py
@@ -31,7 +31,7 @@ class ProcurementException(Exception):
 class Procurement(NamedTuple):
     product_id: fields.Many2one
     product_qty: fields.Float
-    product_uom: fields.Many2one
+    uom_id: fields.Many2one
     location_id: fields.Many2one
     name: fields.Char
     origin: fields.Char
@@ -259,7 +259,7 @@ class StockRule(models.Model):
         final_location_id = False
         if move_to_copy.location_final_id and not move_to_copy.location_dest_id._child_of(move_to_copy.location_final_id):
             final_location_id = move_to_copy.location_final_id.id
-        if move_to_copy.product_uom.compare(move_to_copy.product_uom_qty, 0) < 0:
+        if move_to_copy.uom_id.compare(move_to_copy.product_uom_qty, 0) < 0:
             copied_quantity = move_to_copy.product_uom_qty
         if not company_id:
             company_id = self.sudo().warehouse_id and self.sudo().warehouse_id.company_id.id or self.sudo().picking_type_id.warehouse_id.company_id.id
@@ -295,7 +295,7 @@ class StockRule(models.Model):
                 raise ProcurementException([(procurement, msg)])
 
         # Prepare the move values, adapt the `procure_method` if needed.
-        procurements = sorted(procurements, key=lambda proc: proc[0].product_uom.compare(proc[0].product_qty, 0.0) > 0)
+        procurements = sorted(procurements, key=lambda proc: proc[0].uom_id.compare(proc[0].product_qty, 0.0) > 0)
         for procurement, rule in procurements:
             procure_method = rule.procure_method
             if rule.procure_method == 'mts_else_mto':
@@ -318,7 +318,7 @@ class StockRule(models.Model):
         """
         return []
 
-    def _get_stock_move_values(self, product_id, product_qty, product_uom, location_dest_id, name, origin, company_id, values):
+    def _get_stock_move_values(self, product_id, product_qty, uom_id, location_dest_id, name, origin, company_id, values):
         ''' Returns a dictionary of values that will be used to create a stock move from a procurement.
         This function assumes that the given procurement has a rule (action == 'pull' or 'pull_push') set on it.
 
@@ -346,13 +346,13 @@ class StockRule(models.Model):
                 move_dest.partner_id = self.location_src_id.warehouse_id.partner_id or self.company_id.partner_id
 
         # If the quantity is negative the move should be considered as a refund
-        if product_uom.compare(product_qty, 0.0) < 0:
+        if uom_id.compare(product_qty, 0.0) < 0:
             values['to_refund'] = True
 
         move_values = {
             'company_id': self.company_id.id or self.location_src_id.company_id.id or self.location_dest_id.company_id.id or company_id.id,
             'product_id': product_id.id,
-            'product_uom': product_uom.id,
+            'uom_id': uom_id.id,
             'product_uom_qty': qty_left,
             'partner_id': partner.id if partner else False,
             'location_id': self.location_src_id.id,
@@ -441,7 +441,7 @@ class StockRule(models.Model):
     @api.model
     def _skip_procurement(self, procurement):
         return procurement.product_id.type != "consu" or float_is_zero(
-            procurement.product_qty, precision_rounding=procurement.product_uom.rounding
+            procurement.product_qty, precision_rounding=procurement.uom_id.rounding
         )
 
     @api.model

--- a/addons/stock/models/stock_storage_category.py
+++ b/addons/stock/models/stock_storage_category.py
@@ -58,7 +58,7 @@ class StockStorageCategoryCapacity(models.Model):
             " [('is_storable', '=', True)]"))
     package_type_id = fields.Many2one('stock.package.type', 'Package Type', ondelete='cascade', check_company=True, index='btree_not_null')
     quantity = fields.Float('Quantity', required=True)
-    product_uom_id = fields.Many2one(related='product_id.uom_id')
+    uom_id = fields.Many2one(related='product_id.uom_id')
     company_id = fields.Many2one('res.company', 'Company', related="storage_category_id.company_id")
 
     _positive_quantity = models.Constraint(
